mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBNativeQueryPaginationTest.java

```diff
@@ -0,0 +1,202 @@
+			List result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 0 )
+				.setMaxResults( 10 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 10 );
+		} );
+	}
+
+	@Test
+	public void test_aggregation_match_page1_10() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_AGGREGATE_MATCH )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 0 )
+				.setMaxResults( 10 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 10 );
+		} );
+	}
+
+	@Test
+	public void test_find_page2_5() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 5 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+		} );
+	}
+
+	@Test
+	public void test_aggregation_match_page2_5() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_AGGREGATE_MATCH )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 5 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+		} );
+	}
+
+	@Test
+	public void test_find_page1_page2_5() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 0 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+
+			result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 5 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+		} );
+	}
+
+	@Test
+	public void test_aggregation_match_page1_page2_5() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_AGGREGATE_MATCH )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 0 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+
+			result = session.createNativeQuery( QUERY_AGGREGATE_MATCH )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 5 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+		} );
+	}
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] { OscarWildePoem.class };
+	}
+
+	public void inTransaction(Consumer<Session> consumer) {
+		OgmSession session = openSession();
+		Transaction transaction = session.beginTransaction();
+
+		try {
+			consumer.accept( session );
+			transaction.commit();
+		}
+		catch (Throwable t) {
+			if ( transaction.isActive() ) {
+				transaction.rollback();
+			}
+			throw t;
+		}
+
+		session.clear();
+		session.close();
```

Comment:
Session is autoclosable you can use` try (OgmSession session = openSession) { ... }`, also the session.clear shouldn't be necessary (I think)

Reply:
Could be a nice improvement. Thank you